### ISSUE TITLE
🤔 Some ices get ignored when run non-interactively

### ISSUE THREAD

**Issue by pschmitt** [zdharma-continuum/zinit#97]
**Describe the bug**
When running non-interactively, for eg with Ansible, some ices get ignored. This also happens when run with `zsh -sli`.
My suspicion is that there's something fishy going with zsh hooks (`add-zsh-hook`) in this scenario, since this is how the make ice is called.

This makes it impossible to install *everything* in one go.

**Zinit config**

```zsh
# make does not get executed when running `@zinit-scheduler-burst` w/o a TTY attached
zinit light-mode as"program" make for @direnv/direnv
```

Refs:
- https://github.com/zdharma-continuum/zinit/runs/4275129417?check_suite_focus=true
- #96

**Comment by pschmitt** [comment]
Now I know why this issue seemed familiar:

- https://issueexplorer.com/issue/zdharma/zinit/401
- https://issueexplorer.com/issue/zdharma/zinit/484

TL;DR:
`TERM` needs to be set!
Will fix asap.
